instrumentation/redis/lib/opentelemetry/instrumentation/redis/patches/client.rb

```diff
@@ -48,37 +54,78 @@ def process(commands) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength, M
                   s.record_exception(reply)
                   s.status = Trace::Status.error(reply.message)
                 end
+                next unless config[:record_value_size] && commands.length == 1
+
+                if RETRIEVED_VALUE_SIZE_COMMANDS.include?(commands[0][0])
+                  s['db.retrieved_value_size_bytes'] = calculate_bytesize(reply)
+                end
               end
             end
           end
 
           private
 
+          def calculate_bytesize(value) # rubocop:disable Metrics/CyclomaticComplexity
+            case value
+            when ::Redis::CommandError
+              # Don't report on these; call it zero.
+              0
+            when String
+              value.bytesize
+            when Array
+              value.sum { |s| calculate_bytesize(s) }
+            when Integer
+              value.size
+            when Float
+              # Redis stores floats as strings, so get bytesize of stringified
```

Comment:
Technically, I think it stores floats as floats, but the wire representation is a string. That begs the question: are we interested in the byte size of the wire representation or the stored representation? I think the wire representation is more interesting - the most common analysis is something like "are we sending/receiving a lot of large objects, exhausting network capacity?"

Reply:
Yeah, the comment for sure lacks appropriate rigor. FWIW the Redis docs do not represent `Float` as a [first class data type](https://redis.io/docs/manual/data-types/), but who knows how it's storing these `Float`-y "Strings" behind the scenes. 

When working on this, I was caught off guard by the consideration you bring up - do we care about byte size of wire representation or how much space it takes up on the Redis side (and wait, how does Redis store Floats, and how would we calculate that from Ruby without knowledge of which Redis version etc etc etc). 

I think you're right that we care about the wire representation more than however much disk space Redis is using. Not to mention that figuring out Redis' disk utilization and calculating from this library seems like a bit of a rabbit hole. Also, I've got to imagine that someone running analysis on big keys will end up discovering values that are not `Floats` (or Redis' string representations thereof).